astroid/nodes/node_ng.py

```diff
@@ -227,14 +227,18 @@ def __str__(self) -> str:
 
     def __repr__(self) -> str:
         rname = self.repr_name()
+        try:
+            lineno = self.fromlineno
+        except AttributeError:
```

Comment:
Shall we add a comment like:
`# If the __postinit__ isn't called `fromlineno` could not exist`.

Although I must say, shouldn't `fromlineno` as property always exist?

Reply:
As far as my understanding goes, `fromlineno` does always exist. The problem is that it calls `fixed_source_line` and thats where the `AttributeError` is generated, specifically it seems to come from `get_children`.

I ran this on a different dev machine and got the error for a few other nodes besides `MatchCase` (weird). Here are some sample stack traces:

```
self = <[AttributeError("'Arguments' object has no attribute 'posonlyargs'") raised in repr()] Arguments object at 0x7f4c48347f90>

    def get_children(self):
>       yield from self.posonlyargs or ()
E       AttributeError: 'Arguments' object has no attribute 'posonlyargs'
```

```
@cached_property
def fromlineno(self) -> int:
....
>       if self.decorators is not None:
E       AttributeError: 'AsyncFunctionDef' object has no attribute 'decorators'
```

I guess the comment could be something like `The dependencies used to calculate fromlineno (if not cached) may not exist at the time`?

Reply:
Adding to my previous comment, in more general terms it seems `fromlineno` may execute certain function calls in an attempt to obtain the line number, and the resources used in said functions may not exist at the time of the read attempt.

Reply:
Yeah you suggestion for the comment seems fine! Thanks!